driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -509,6 +502,16 @@ private <T> void sendCommandMessageAsync(final int messageId, final Decoder<T> d
                                              final SingleResultCallback<T> callback, final ByteBufferBsonOutput bsonOutput,
                                              final CommandEventSender commandEventSender, final boolean responseExpected) {
         List<ByteBuf> byteBuffers = bsonOutput.getByteBuffers();
+
+        boolean[] shouldReturn = {false};
```

Comment:
I've made changes to move the log message from "after the write is complete" to "before the write is complete."

According to the CSOT spec:
> After wire message construction, drivers MUST check for timeout before writing the message to the server.